daemon/containerd/service.go

```diff
@@ -154,9 +156,35 @@ func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID st
 	if ctr == nil {
 		return 0, 0, nil
 	}
+
+	snapshotter := i.client.SnapshotService(ctr.Driver)
+
+	usage, err := snapshotter.Usage(ctx, containerID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	imageManifest, err := getContainerImageManifest(ctr)
+	if err != nil {
+		// Best efforts attempt to pick an image.
+		// We don't have platform information at this point, so we can only
+		// assume that the platform matches host.
+		// Otherwise this will give a wrong base image size (different
+		// platform), but should be close enough.
+		mfst, err := i.GetImageManifest(ctx, ctr.Config.Image, imagetypes.GetImageOpts{})
+		if mfst == nil || err != nil {
```

Comment:
Do we know in which case it returns a `nil` manifests _without_ an (notfound?) error?

Reply:
It doesn't in any known to me case. Just wanted to be explicit about not dereferencing a nil pointer.

Reply:
Should we let it panic in that case (if we regard it as a bug, a panic is very visible), or just log other error?

Reply:
I _suspect_ it would already panic if it would be `nil`; looking at 

https://github.com/moby/moby/blob/801202349ab04a329c32da5612f230689c54eb91/daemon/containerd/image.go#L159-L161


and

https://github.com/moby/moby/blob/801202349ab04a329c32da5612f230689c54eb91/daemon/containerd/image.go#L179

😅

Reply:
Removed the mfst == nil check